solutions/web3-data-fetching/pages/index.tsx

```diff
@@ -0,0 +1,123 @@
+import Head from 'next/head'
+import { Layout, Text, Page, Link } from '@vercel/examples-ui'
+import { ethers } from 'ethers'
+import useSWR from 'swr'
+import { FC } from 'react'
+import abi from '../lib/BAYC.abi.json'
+import { BORED_APE_YATCH_CLUB_ADDRESS } from '../constants'
+
+const contract = new ethers.Contract(
+  BORED_APE_YATCH_CLUB_ADDRESS,
+  abi,
+  ethers.getDefaultProvider()
+)
+
+const Snippet: FC = ({ children }) => {
+  return (
+    <pre className="border-accents-2 border rounded-md bg-white overflow-x-auto p-6 transition-all">
+      {children}
+    </pre>
+  )
+}
+
+function Home({ contractName }: { contractName: string }) {
+  const { data } = useSWR('name', () => contract.name())
+
+  return (
+    <Page>
+      <Head>
+        <title>Data from smart contracts - Vercel Example</title>
+        <meta
+          name="description"
+          content="Vercel example how to use data from smart contracts"
+        />
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+
+      <section className="flex flex-col gap-6">
+        <Text variant="h1">Fetching data from smart contracts</Text>
+        <Text>
+          Smart contracts contain relevant information to applications built on
+          top of blockchains that can run the{' '}
+          <Link href="https://ethereum.org/en/developers/docs/evm/">
+            Ethereum Virtual Machine
+          </Link>
+          . Some of these contracts informations can be exposed in the form of
+          View functions that do not need{' '}
+          <Link href="https://ethereum.org/en/developers/docs/gas/">
+            {' '}
+            gas or fees
+          </Link>{' '}
+          to be executed. Now we will explore how to get that information in
+          Next.js
+        </Text>
+
+        <Text variant="h2">
+          Instanciating a connection to the smart contract
+        </Text>
+        <Text>
+          A first step needed to contact these smart contract via RPC is to
+          instantiate a connection to them using a library like{' '}
+          <Link href="https://docs.ethers.io/v5/">Ethers.js</Link>. There are
+          also convenience libraries like{' '}
+          <Link href="https://github.com/dethcrypto/TypeChain">Typechain</Link>{' '}
+          to help in this process.
+        </Text>
+        <Text>
+          The ABI contains information about the available function. Get it from
+          Etherscan like explained{' '}
+          <Link href="https://cryptomarketpool.com/how-to-get-a-smart-contracts-abi-for-use-in-python-web3-py/">
+            Here
+          </Link>
```

Comment:
Let's change this to:

> The ABI contains information about the available function and [can be obtained through Etherscan](https://cryptomarketpool.com/how-to-get-a-smart-contracts-abi-for-use-in-python-web3-py/). We will use the Bored Ape Yatch Club popular NFT contract.

The reason is that links with "here" don't have information about the place where you're taking the visitor to.